# HG changeset patch
# User shelve@localhost
# Date 1659537748 21600
#      Wed Aug 03 08:42:28 2022 -0600
# Branch quic
# Node ID d7c7377ce87834b995f10337c40f5c331b9fac1a
# Parent  3550b00d9dc8a307dbaa1aeb9f988cec2a08476a
changes to: QUIC: avoided pool usage in token calculation.

diff --git a/src/event/quic/ngx_event_quic_ack.c b/src/event/quic/ngx_event_quic_ack.c
--- a/src/event/quic/ngx_event_quic_ack.c
+++ b/src/event/quic/ngx_event_quic_ack.c
@@ -333,14 +333,14 @@
     }
 
     if (cg->window < cg->ssthresh) {
-        cg->window += f->plen;
+        cg->window += f->plen / 2;
 
         ngx_log_debug3(NGX_LOG_DEBUG_EVENT, c->log, 0,
                        "quic congestion slow start win:%uz ss:%z if:%uz",
                        cg->window, cg->ssthresh, cg->in_flight);
 
     } else {
-        cg->window += qc->tp.max_udp_payload_size * f->plen / cg->window;
+        cg->window += (qc->tp.max_udp_payload_size * f->plen / cg->window) / 2;
 
         ngx_log_debug3(NGX_LOG_DEBUG_EVENT, c->log, 0,
                        "quic congestion avoidance win:%uz ss:%z if:%uz",
